Runtime/CesiumIntMatN.cs

```diff
@@ -0,0 +1,395 @@
+using System.Diagnostics;
+using System;
+using Unity.Mathematics;
+
+namespace CesiumForUnity
+{
+    /// <summary>
+    /// Represents a mat2x2 with signed integer components. This preserves the accuracy of the integer
+    /// values, which could otherwise lose precision if represented as doubles.
```

Comment:
FWIW, only 64-bit integers might lose precision. All 8, 16, and 32-bit integers can be represented perfectly as doubles.

Reply:
That's good to know! I still prefer having `CesiumIntMatN` because it stores a component type, indicating whether its values are meant to be interpreted as `Int8`, `Int16`, `Int32`, or `Int64`, and that gets lost if represented as a Unity `dmat2x2`. But I'll modify the comment.